cloudify_agent/api/plugins/installer.py

```diff
@@ -135,67 +140,98 @@ def _install_managed_plugin(self,
         dst_dir = self._full_dst_dir(dst_dir, managed_plugin)
         self.logger.debug('Checking if managed plugin installation exists '
                           'in {0}'.format(dst_dir))
-        lock = self._lock(dst_dir)
-        lock.acquire()
-        try:
-            if os.path.exists(dst_dir):
-                self.logger.debug('Plugin path exists {0}'.format(dst_dir))
-                plugin_id_path = os.path.join(dst_dir, 'plugin.id')
-                if os.path.exists(plugin_id_path):
-                    self.logger.debug('Plugin id path exists {0}'.
-                                      format(plugin_id_path))
-                    with open(plugin_id_path) as f:
-                        existing_plugin_id = f.read().strip()
-                    matching_existing_installation = (
-                        existing_plugin_id == managed_plugin.id)
-                    if not matching_existing_installation:
-                        raise exceptions.PluginInstallationError(
-                            'Managed plugin installation found but its ID '
-                            'does not match the ID of the plugin currently '
-                            'on the manager. [existing: {0}, new: {1}]'
-                            .format(existing_plugin_id,
-                                    managed_plugin.id))
-                else:
+
+        # create_deployment_env should wait for the install_plugin wf to finish
+        while (self._is_plugin_installing(managed_plugin.id) and
+               deployment_id != SYSTEM_DEPLOYMENT):
+            time.sleep(1)
```

Comment:
magic number, perhaps export to a constant?
Also, it's worth checking if checking the DB every 1 second is a good idea.